# ML-02 — Research Question and Provisional Lane

[![Open In Colab](https://colab.research.google.com/assets/colab-badge.svg)](https://colab.research.google.com/github/YoyuQre/flyrank_assgn_1/blob/main/work/notebooks/w01_research_question.ipynb?flush_cache=true)

This skeleton is yours to fill. Work the sections **in order** — each one has a one-line hint. Simple words, honest numbers.

> Working with an AI assistant? Tell it to read `skills/README.md` first and load the one skill this assignment names on its card.

## 1. My lane (or freestyle) and why

*Name your lane — or say 'freestyle' and describe your own question. One short paragraph: why this one?*

**Content Refresh Prediction** lane because it addresses a practical SEO challenge: identifying webpages that are likely to experience a decline in search performance so they can be reviewed and refreshed before significant traffic is lost. This problem has clear business value, historical search-performance data is available to learn meaningful patterns, and the predictions can help content editors and SEO teams prioritize pages that are most likely to benefit from timely updates.


In [8]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.

## 2. The question: decision, action, cost of a wrong call

*What decision does your work improve? Who acts on it? What does a wrong recommendation cost?*

This project improves the decision of **which webpages should be prioritized for review and content refresh before their search performance declines significantly**. The primary users are **SEO specialists and content editors**, who will use the model's recommendations to decide where to focus their limited time and resources. A **false positive** may lead to unnecessary effort spent reviewing a page that was unlikely to decline, while a **false negative** is more costly because a genuinely declining page may be overlooked, resulting in reduced search visibility, lower organic traffic, and missed opportunities to improve performance.


In [9]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## 3. Quick look at the data (2-3 real numbers)

*Load the starter CSV below and show 2-3 real numbers that make your lane look worth the next 7 weeks.*

In [10]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.
import pandas as pd
df=pd.read_csv('content_refresh_anonymized.csv')
print(f"Total Content Pages: {len(df):,}")
print(f"Average impressions (last 90 days):{df['impressions_90d'].mean():.1f}")
print(f"Average days since last update::{df['days_since_last_update'].mean():.1f}")
print("\nTrend distribution:")
print(df['trend_direction'].value_counts())
print("\nPercentage by trend:")
print((df['trend_direction'].value_counts(normalize=True) * 100).round(2))

Total Content Pages: 30,000
Average impressions (last 90 days):5200.4
Average days since last update::46.1

Trend distribution:
trend_direction
down      16262
stable     5962
up         4388
new        2236
flat       1152
Name: count, dtype: int64

Percentage by trend:
trend_direction
down      54.21
stable    19.87
up        14.63
new        7.45
flat       3.84
Name: proportion, dtype: float64


## 4. Careful words: what I can and can't claim

*Write what your work will be able to say (observed, directional, decision-support) — and what it never will (causal proof, 'predicting Google').*

The starter dataset contains **30,000 content pages**, providing a substantial amount of historical search-performance data for analysis. On average, each page received **5,200.4 impressions** over the last 90 days and was last updated **46.1 days** ago, indicating variation in both visibility and content freshness. Most importantly, **54.21%** of the pages are labeled with a **downward trend**, suggesting that content performance changes frequently and making **Content Refresh Prediction** a meaningful problem. These observations indicate that there is sufficient data and a practical business need to develop a model that helps prioritize pages for timely content updates.


In [11]:
# This cell is for CODE (numbers, a query, a check).
# Write your text answer in the cell ABOVE this one — typing sentences here breaks Run All.


## Self-check

Before you submit, confirm each line honestly:

- [ ] Every section above is filled — markdown thinking AND the code that backs it
- [ ] The notebook runs top to bottom with no errors (Runtime → Run all)
- [ ] No client names, URLs, or private queries anywhere
- [ ] My claims use careful words: observed, measured, directional, decision-support
- [ ] Committed to my repo under `work/notebooks/` — then submit your repo URL on the card. Done.